### PR TITLE
server: resign leader if leader not changed.

### PR DESCRIPTION
**Pull request by siddontang**
This can only occur when we meet error with Etcd, and it is hard to write test now. 
I will do some abstraction and let test with etcd easily. 

@qiuyesuifeng @disksing

### CODE REVIEW

**Comment by disksing** [comment]
LGTM

File: server/leader.go [hunk 1]
```diff
@@ -70,13 +70,24 @@ func (s *Server) leaderLoop() {
...
-			s.watchLeader()
-			log.Info("leader changed, try to campaign leader")
+			if s.isSameLeader(leader) {
+				// oh, we are already leader, we may meet something wrong
+				// in previous campaignLeader. we can resign and campaign again.
+				log.Warnf("leader is still %s, resign and campaign again", leader)
+				if err = s.resignLeader(); err != nil {
+					log.Errorf("resign leader err %s", err)
+					time.Sleep(200 * time.Millisecond)
+					continue
```
Comment (shenli) [inline server/leader.go]: Set a retry count limitation?

Comment (siddontang) [inline server/leader.go]: no need, outer will retry again.

**Comment by shenli** [comment]
LGTM

**Pull request merged**